pkg/controller/kogitodataindex/kogitodataindex_controller.go

```diff
@@ -146,7 +146,7 @@ func (r *ReconcileKogitoDataIndex) Reconcile(request reconcile.Request) (reconci
 	// Deploy infra dependencies
 	if result, err := r.ensureKogitoInfra(instance); err != nil {
 		return reconcile.Result{}, err
-	} else if result != nil {
+	} else if result != nil && result.RequeueAfter > 0 {
```

Comment:
Updating the `instance` in `ensureKogitoInfra` is supposed to trigger a new iteration of `reconcile`.
https://github.com/kiegroup/kogito-cloud-operator/blob/ac0d2dbbc82d6dc310736ebc5a55c07e8cd95e63/pkg/controller/kogitodataindex/kogitodataindex_controller.go#L191
In this case, the current iteration of `reconcile` should return without requeue, and let the new iteration of `reconcile` to handle changes.
I also created a fix to this issue in pull request #133 . https://github.com/kiegroup/kogito-cloud-operator/blob/1ca5a3b3ed9c77f331405bd7aa556d1a34d3bb79/pkg/controller/kogitodataindex/kogitodataindex_controller.go#L232

Reply:
Yes, you're right. The trick part was the `return nil, nil`. Thanks! Updated.